### PR TITLE
Update load-tests.md

### LINKED ISSUE #22430: Add don't perf test in dev mode
per https://github.com/dotnet/aspnetcore/issues/32977

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ce92350f-25e5-63bc-c815-a8a41b2253c3
* Version Independent ID: e8f9ec21-4e0a-054e-b9ba-fd27e71e31a0
* Content: [ASP.NET Core load/stress testing](https://docs.microsoft.com/en-us/aspnet/core/test/load-tests?view=aspnetcore-5.0)
* Content Source: [aspnetcore/test/load-tests.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/test/load-tests.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-test**
* GitHub Login: @Jeremy-Meng
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Address https://github.com/dotnet/aspnetcore/issues/32977 and Fixes #22430

### CODE REVIEW

**Comment by davidfowl** [comment]
Unfortunately, this alone doesn't fix it. You need to also set the environment to Production. Running in Visual Studio sets this to Development by default (and it has nothing to do with Release and Debug)

File: aspnetcore/test/load-tests.md [hunk 1]
```diff
@@ -33,3 +33,7 @@ The following list contains third-party web performance tools with various featu
 * [Netling](https://github.com/hallatore/Netling)
 * [Vegeta](https://github.com/tsenart/vegeta)
 * [NBomber](https://github.com/PragmaticFlow/NBomber)
+
+## Load and stress test with release builds
+
+Load and stress tests should be done in release and [production](xref:fundamentals/environments) mode and not in debug or development mode. [Release configurations](/visualstudio/debugger/how-to-set-debug-and-release-configurations) are fully optimized with minimal logging. Debug configuration is not optimized and has logging that can impact performance.
```
Comment (Rick-Anderson) [inline aspnetcore/test/load-tests.md:39]:
```suggestion
Load and stress tests should be done in release and [production](xref:fundamentals/environments) mode and not in debug and development mode. [Release configurations](/visualstudio/debugger/how-to-set-debug-and-release-configurations) are fully optimized with minimal logging. Debug configuration is not optimized.   [Development](xref:fundamentals/environments) mode enables more information logging that can impact performance.
```

**Comment by Rick-Anderson** [comment]
@mkArtakMSFT please suggest a reviewer

File: aspnetcore/test/load-tests.md [hunk 2]
```diff
@@ -33,3 +33,7 @@ The following list contains third-party web performance tools with various featu
 * [Netling](https://github.com/hallatore/Netling)
 * [Vegeta](https://github.com/tsenart/vegeta)
 * [NBomber](https://github.com/PragmaticFlow/NBomber)
+
+## Load and stress test with release builds
+
+Load and stress tests should be done in release and [production](xref:fundamentals/environments) mode and not in debug and development mode. [Release configurations](/visualstudio/debugger/how-to-set-debug-and-release-configurations) are fully optimized with minimal logging. Debug configuration is not optimized.  [Development](xref:fundamentals/environments) mode enables more information logging that can impact performance.
```
Comment (Rick-Anderson) [inline aspnetcore/test/load-tests.md:39]:
```suggestion
Load and stress tests should be done in release and [production](xref:fundamentals/environments) mode and not in debug and development mode. [Release configurations](/visualstudio/debugger/how-to-set-debug-and-release-configurations) are fully optimized with minimal logging. Debug configuration is not optimized. [Development](xref:fundamentals/environments) mode enables more information logging that can impact performance.
```

**Review by mjrousos — APPROVED** [review]
LGTM

**Pull request merged**